tripal/includes/tripal.registration.inc

```diff
@@ -0,0 +1,336 @@
+      '#title' => t('Grant Number'),
+    );
+    $form['details']['funding'][$i]['tripal_reg_site_amount'] = array(
+      '#type' => 'textfield',
+      '#default_value' => isset($form_data['values']['funding'][$i]['tripal_reg_site_amount']) ? $form_data['values']['funding'][$i]['tripal_reg_site_amount'] : null,
+      '#title' => t('Funding Amount'),
+    );
+
+    $form['details']['funding'][$i]['funding_period'] = array(
+      '#type' => 'fieldset',
+      '#title' => t('Funding Period'),
+      '#tree' => TRUE,
+    );
+
+    $form['details']['funding'][$i]['funding_period']['tripal_reg_site_start'] = array(
+      '#type' => 'date_select',
+      '#title' => t("Start"),
+      '#default_value' => isset($form_data['values']['funding'][$i]['funding_period']['tripal_reg_site_start']) ? $form_data['values']['funding'][$i]['funding_period']['tripal_reg_site_start'] : date('Y', time()),
+      '#date_year_range' => '-20:+20',
+      '#date_format' => 'Y',
+    );
+
+    $form['details']['funding'][$i]['funding_period']['tripal_reg_site_end'] = array(
+      '#type' => 'date_select',
+      '#title' => t('End'),
+      '#default_value' => isset($form_data['values']['funding'][$i]['funding_period']['tripal_reg_site_end']) ? $form_data['values']['funding'][$i]['funding_period']['tripal_reg_site_end'] : date('Y', time()),
+      '#date_year_range' => '-20:+20',
+      '#date_format' => 'Y',
+    );
+  }
+  
+  $form['details']['funding']['add_funding'] = array(
+    '#type' => 'button',
+    '#value' => t('Add additional funding sources'),
+    '#href' => '',
+    '#ajax' => array(
+      'callback' => 'custom_registration_ajax_add_funding',
+      'wrapper' => 'funding',
+    ),
+  );
+  $form_state['details']['funding']++;
+
+  // Provide a submit button.
+  if (!empty($form_data)) {
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#value' => 'Update registration information',
+    );
+  }
+  else {
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#value' => 'Register Site',
+    );
+  }
+
+
+  return $form;
+
+}
+
+function custom_registration_ajax_add_funding($form, $form_state) {
+  return $form['details']['funding'];
+}
+
+function _tripal_form_disable_reg_callback($form, &$form_state) {
+  variable_set('disable_tripal_reporting', TRUE);
+}
+
+/**
+ * Implements validation from the Form API.
+ * 
+ * @param $form
+ *   A structured array containing the elements and properties of the form.
+ * @param $form_state
+ *   An array that stores information about the form's current state 
+ *   during processing.
+ */
+function tripal_registration_form_validate($form, &$form_state){
+  $mail_pi = $form_state['values']['principal_investigator_email'];
+  $mail_sa = $form_state['values']['tripal_reg_site_admin_email'];
+  if (!valid_email_address($mail_pi)) {
```

Comment:
I get this error when submitting  no PI email.  However the wording implies its not a required field.  If its required, there needs to be a way to opt out of email notices.  If it isnt required, then dont run `valid_email_address` if the value is null

![Screen Shot 2019-04-01 at 3 45 11 PM](https://user-images.githubusercontent.com/7063154/55355290-dad89c80-5495-11e9-854e-55099dafff4a.png)

Reply:
I no longer see this error on testing :-)